core/src/main/res/drawable/custom_dialog_shape_drawable_left_off.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android" >
```

Comment:
Hmm now that I think about it, this will lead to a lot of different drawables. All combinations of on/off, left/right, light/dark. This means there will be at least 8 files just for handling the backgrounds. If you use a `CardView` with `app:borderRadius`, you can achieve something like the following without custom drawables. I think this makes the code easier to understand. I know that it looks a bit different than the initial drawing but I think that it is worth the change because it makes the code a lot shorter and easier to understand.

![image](https://user-images.githubusercontent.com/5811634/83980919-a2deac00-a919-11ea-9e20-8b1379385a30.png)

Reply:
CardView ok -so we are removing the blue border and this is the design now

Reply:
Sorry for requesting so many changes and causing you additional work. I just try to keep the code base as clean as possible and realized that having 8 copies of basically the same file will probably lead to issues and inconsistencies sooner or later.

Reply:
> so we are removing the blue border and this is the design now

Hmm. With the new set-up it's not possible to have a line? Maybe, then, in that case, should it be two separate 'blubs' instead (so that the buttons are clearly separated, while still 'interacting' with each-other)?

Reply:
The outer lines can only be done using lots of images while just rounding can be done using CardView. I think separate buttons could introduce the problem that it is harder to see that they interact with each other. If you worry about a separator when both buttons are de-selected, this can still be done by moving the buttons apart with margin (within the CardView).

Reply:
No problem , just ran out of time last week. I think it's a good idea. From what I understand we're going to have 3 images a left-active, right-active and a "x" deselected empty active state.

Reply:
> From what I understand we're going to have 3 images

My idea was to have no image at all for the backgrounds. Everything is added to a CardView for rounding and then we just set the background of the RadioButtons using a color selector for on/off state (https://stackoverflow.com/a/3565624/4193263).

Reply:
okay

Reply:
I've gone through the UI portion, I haven't added the enum update yet...
![not_good](https://user-images.githubusercontent.com/262625/84453951-3d473400-ac27-11ea-9cf5-ff086e1d8991.png)
If I use the method you were thinking about, the card shape backgrounds will be similar to them example above and I'm going to have to put 3 cardviews per row for it to operate anything similar to what you want, which doesn't make sense to me. There is no borderRadiusLEFT or RIGHT borderRadius covers all corners for a card background as far as I know. Even though the design you want is simple it seems more complicated than what the cardView is currently made for.

I tried getting something working as close to what I think you want as I could while looking exactly how you want it with only 3 different drawables. Three files loading into a background image of the cardview.

filter_dialog_2_clear.xml
filter_dialog_2_left_on.xml
filter_dialog_2_right_on.xml

![new_filter_dialog_screenshot](https://user-images.githubusercontent.com/262625/84454297-22c18a80-ac28-11ea-9633-adacc0c12491.png)

I can't find a lighter shade of grey in the colors that are already specified. 

If you decide later you want 3 elements per row you just need to update in 3 different layout files and the filterdialog radiobuttons.

Reply:
> If I use the method you were thinking about, the card shape backgrounds will be similar to the example above and I'm going to have to put 3 cardviews per row for it to operate anything similar to what you want

You can use one single CardView to round the outer shape and then add RadioButtons with simple color backgrounds (no images) like below. My example is missing the color state selector but I hope that it is now more clear what I meant.

![image](https://user-images.githubusercontent.com/5811634/84478885-546f3b80-ac92-11ea-98ca-249a6a47377f.png)

    <androidx.cardview.widget.CardView
            android:layout_width="match_parent"
            android:layout_height="32dp"
            app:cardCornerRadius="16dp"
            app:cardElevation="0dp"
            android:clipChildren="true">

        <RadioGroup
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                android:orientation="horizontal">

            <RadioButton
                    android:layout_width="wrap_content"
                    android:layout_height="match_parent"
                    android:layout_weight="1"
                    android:background="#ddd"
                    android:layout_marginRight="2dp"
                    android:layout_marginEnd="2dp"
                    android:text="Text1"
                    android:gravity="center"
                    android:button="@android:color/transparent" />

            <RadioButton
                    android:layout_width="wrap_content"
                    android:layout_height="match_parent"
                    android:layout_weight="1"
                    android:background="#ddd"
                    android:text="Text2"
                    android:gravity="center"
                    android:button="@android:color/transparent" />

        </RadioGroup>

    </androidx.cardview.widget.CardView>

Reply:
okay we've eliminated the confusion it seems
```
<CardView>
    <RadioGroup>
        <RadioButton />
        <RadioButton />
        ...
        <RadioButton />
        <ImageView /> <--------- where you lost me
    </RadioGroup>
</CardView>
```

That's a little easier isn't it.

What about the clear button. Do you want the drawable for the on/off state or do you want that simple background as well?

Reply:
> where you lost me

You are right. What I wrote is wrong and confusing. What about using something like this?

```xml
<RadioGroup>
    <CardView>
        <LinearLayout>
            <RadioButton />
            <RadioButton />
            ...
            <RadioButton />
        </LinearLayout>
    </CardView>
    <RadioButton /> <!-- Clear button -->
</RadioGroup>
```

I would use a single drawable for the clear button but use `android:buttonTint` together with a color state selector to achieve the effect with as few images as possible.

Reply:
okay